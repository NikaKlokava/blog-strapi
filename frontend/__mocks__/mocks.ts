import image1 from "../public/test_img1.jpg";
import image2 from "../public/test_img2.jpg";
import image3 from "../public/test_img3.jpg";
import image4 from "../public/test_img4.jpg";
import image5 from "../public/test_img5.jpg";
import image6 from "../public/test_img6.jpg";
import back from "../public/back.jpg";

import instagram from "../public/instagram.svg";
import linkedin from "../public/linkedin.svg";
import github from "../public/github.svg";

export const mockLinks: FooterLinks = [
  {
    id: "asb",
    attributes: {
      name: "linkedin",
      href: "https://www.linkedin.com/in/veranika-klokava-858b5b287",
      svg: linkedin,
    },
  },

  {
    id: "dfsdf",
    attributes: {
      name: "github",
      href: "https://github.com/NikaKlokava",
      svg: github,
    },
  },
  {
    id: "dff",
    attributes: {
      name: "instagram",
      href: "https://www.instagram.com/nika_klokava",
      svg: instagram,
    },
  },
];

export const postsMockData: Post[] = [
  {
    section: "styles",
    title: "Style title example 3",
    date: "16 October 2022",
    photo: image3,
    best_status: true,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image2, image3, image6, image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image6, image1, image2, image3],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image2, image6, image1, image3, image6, image1],
          },
        },
      ],
    },
  },
  {
    section: "travel",
    title: "Travel title example 2",
    date: "16 October 2022",
    photo: image1,
    best_status: true,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image6, image1],
          },
        },
      ],
    },
  },
  {
    section: "styles",
    title: "Style title example ",
    date: "16 October 2022",
    photo: image3,
    best_status: false,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image5],
          },
        },
      ],
    },
  },
  {
    section: "styles",
    title: "Styles title example 2",
    date: "16 October 2022",
    photo: image3,
    best_status: false,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image5],
          },
        },
      ],
    },
  },
  {
    section: "life",
    title: "Life title example ",
    date: "16 October 2022",
    photo: image2,
    best_status: false,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image1, image4],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna.`,
            photos: [image1, image5, image2],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image5],
          },
        },
      ],
    },
  },
  {
    section: "life",
    title: "Life title example 2",
    date: "16 October 2022",
    photo: image2,
    best_status: false,
    post_contents: {
      data: [
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image1, image4],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image1],
          },
        },
        {
          id: 555,
          attributes: {
            title: "some title",
            description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
            photos: [image3, image5],
          },
        },
      ],
    },
  },
];

export const aboutMockData: AboutData = {
  id: 12345,
  attributes: {
    greeting: "Welcome to my blog!",
    description: `At vero eos et accusamus et iusto odio dignissimos
      ducimus qui blanditiis praesentium voluptatum deleniti
      atque corrupti quos dolores et quas molestias excepturi sint occaecati
      cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia
      animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est
      et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio
      cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas
      assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis
      aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non
      recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis
      voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat`,
    photo: image4,
  },
};

export const wishListMockData: WishListData = Array(10).fill({
  id: "123",
  attributes: {
    href: "https://www.zara.com/lt/lt/%E2%80%9Ezw-collection--pavilkinio-stiliaus-suknele-su-zvyneliais-p09204110.html?v1=307074382&v2=2317646",
    photo: image5,
  },
});

export const instaAestheticData: AestheticsData = Array(9).fill({
  id: 3456,
  attributes: {
    photo: image6,
    description: `Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi`,
    date: "14th October 2023",
    location: "Vancouver",
    tags: {
      data: [
        {
          id: 222,
          attributes: { tag: "#travel" },
        },
        {
          id: 222,
          attributes: { tag: "#aesthetics" },
        },
        {
          id: 222,
          attributes: { tag: "#life" },
        },
        {
          id: 222,
          attributes: { tag: "#blog" },
        },
      ],
    },
  },
});

export const homeMockData: HomeData = {
  id: 123,
  attributes: {
    title: "Hello from TRAVELER!",
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Temporibus
  fugit, aliquam animi nostrum repudiandae non quibusdam. Commodi,
  blanditiis, mollitia eaque reprehenderit fugiat, aut facere dolorem
  voluptatum a nulla ut. Iure? Lorem ipsum dolor sit amet consectetur
  adipisicing elit. Temporibus fugit, aliquam animi nostrum repudiandae
  non quibusdam. Commodi, blanditiis, mollitia eaque reprehenderit fugiat,
  aut facere dolorem voluptatum a nulla ut. Iure?`,
    background: back,
  },
};
