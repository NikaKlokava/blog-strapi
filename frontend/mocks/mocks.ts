import image1 from "../public/test_img1.jpg";
import image2 from "../public/test_img2.jpg";
import image3 from "../public/test_img3.jpg";
import image4 from "../public/test_img4.jpg";
import image5 from "../public/test_img5.jpg";
import image6 from "../public/test_img6.jpg";

export const links = [
  {
    name: "linkedin",
    href: "https://www.linkedin.com/in/veranika-klokava-858b5b287",
  },
  { name: "github", href: "https://github.com/NikaKlokava" },
  { name: "instagram", href: "https://www.instagram.com/nika_klokava" },
];

export const navbarItems = [
  {
    name: "HOME",
    path: "/",
  },
  {
    name: "TRAVEL",
    path: "/sections/travel",
  },
  {
    name: "LIFE",
    path: "/sections/life",
  },
  {
    name: "STYLES",
    path: "/sections/styles",
  },

  {
    name: "THE BEST",
    path: "/the-best",
  },
  {
    name: "INSTA AESTHETICS",
    path: "/insta-aesthetics",
  },
  {
    name: "ABOUT",
    path: "/about",
  },
];
type Data = {
  id: string;
  name: string;
  descr: string;
  title: string;
  date: string;
  photo: any;
  content: {
    description: string;
    photos: any[];
  }[];
};

export const sectionsData: Data[] = [
  {
    id: "123dsaaa",
    name: "travel",
    descr: "travel_img",
    title: "Some title example",
    date: "16 October 2022",
    photo: image1,
    content: [
      {
        description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
        photos: [image2, image3, image6, image1],
      },
    ],
  },
  {
    id: "1234dvcad",
    name: "life",
    descr: "life_img",
    title: "Some title example",
    date: "16 October 2022",
    photo: image2,
    content: [
      {
        description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
        photos: [image2, image3, image6, image1],
      },
    ],
  },
  {
    id: "123sdvadsa",
    name: "styles",
    descr: "styles_img",
    title: "Some title example",
    date: "16 October 2022",
    photo: image3,
    content: [
      {
        description: `Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna`,
        photos: [image2, image3, image6, image1],
      },
    ],
  },
];

// export const sectionsData: Data[] = [
//   {
//     name: "travel",
//     logo: logo1,
//     data: Array(10).fill({
//       name: "travel",
//       descr: "travel_img",
//       title: "Some title example",
//       date: "16 October 2022",
//       photo: image1,
//       content: [
//         {
//           description: `Lorem ipsum dolor sit amet, consectetur
//           adipiscing elit, sed do eiusmod tempor incididunt ut
//           labore et dolore magna`,
//           photos: [image2, image3, image6, image1],
//         },
//       ],
//     }),
//   },
//   {
//     name: "life",
//     logo: logo2,
//     data: Array(18).fill({
//       name: "life",
//       descr: "travel_img",
//       title: "Some title example",
//       date: "14 October 2022",
//       photo: image2,
//     }),
//   },
//   {
//     name: "styles",
//     logo: logo3,
//     data: Array(15).fill({
//       name: "styles",
//       descr: "travel_img",
//       title: "Some title example",
//       date: "15 October 2022",
//       photo: image3,
//     }),
//   },
// ];

export const aboutData = {
  greeting: "Welcome to my blog!",
  description: `At vero eos et accusamus et iusto odio dignissimos 
    ducimus qui blanditiis praesentium voluptatum deleniti 
    atque corrupti quos dolores et quas molestias excepturi sint occaecati 
    cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia 
    animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est 
    et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio 
    cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas 
    assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis 
    aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non 
    recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis 
    voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat`,
  photo: image4,
};

type WishListType = {
  href: string;
  photo: any;
};
export const wishListData: WishListType[] = Array(10).fill({
  href: "https://www.zara.com/lt/lt/%E2%80%9Ezw-collection--pavilkinio-stiliaus-suknele-su-zvyneliais-p09204110.html?v1=307074382&v2=2317646",
  photo: image5,
});

export type InstaAestheticType = {
  photo: any;
  description: string;
  date: string;
  location: string;
  tags: string[];
};

export const instaAestheticData: InstaAestheticType[] = Array(9).fill({
  photo: image6,
  description: `Lorem ipsum dolor sit amet, consectetur
  adipiscing elit, sed do eiusmod tempor incididunt ut
  labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco laboris nisi`,
  date: "14th October 2023",
  location: "Vancouver",
  tags: ["#travel", "#aesthetics", "#life", "#blog"],
});
