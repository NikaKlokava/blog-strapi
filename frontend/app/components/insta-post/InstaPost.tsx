"use client";

import { InstaAestheticType } from "@/mocks/mocks";
import Image from "next/image";
import styles from "./InstaPost.module.css";

type Props = {
  post: InstaAestheticType;
  close: () => void;
};

export const InstaPost = ({ post, close }: Props) => {
  return (
    <>
      <div className={styles.modal_wrapper} onClick={close}></div>
      <div className={styles.modal_container}>
        <Image
          src={post?.photo}
          alt={"current_post"}
          // width={200}
          // height={200}
          priority={true}
          className={styles.post_photo}
        />
        <div className={styles.post_data_container}>
          <div className={styles.close_icon} onClick={close} />
          <div className={styles.location}>{post?.location}</div>
          <div className={styles.date}>{post?.date}</div>
          <div className={styles.post_descripton}>{post?.description}</div>
          <div className={styles.tags_container}>
            {post?.tags.map((tag, i) => {
              return <div key={i}>{tag}</div>;
            })}
          </div>
        </div>
      </div>
    </>
  );
};
