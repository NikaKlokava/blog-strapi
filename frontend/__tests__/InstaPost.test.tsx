import { instaAestheticData } from "@/__mocks__/mocks";
import { InstaPost } from "@/app/components/insta-post/InstaPost";
import { render } from "@testing-library/react";

describe("Test the InstaPost component", () => {
  it("The InstaPost component renders correctly", () => {
    const instaPostSnap = render(
      <InstaPost
        post={instaAestheticData[0].attributes}
        close={() => console.log("test")}
      />
    );
    expect(instaPostSnap).toMatchSnapshot();
  });
});
