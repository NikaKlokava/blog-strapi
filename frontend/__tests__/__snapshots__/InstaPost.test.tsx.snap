// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the InstaPost component The InstaPost component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal_wrapper"
      />
      <div
        class="modal_container"
      >
        <img
          alt="current_post"
          class="post_photo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="400"
          src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="450"
        />
        <div
          class="post_data_container"
        >
          <img
            alt="close_icon"
            class="close_icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="location"
          >
            Vancouver
          </div>
          <div
            class="date"
          >
            14th October 2023
          </div>
          <div
            class="post_descripton"
          >
            Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi
          </div>
          <div
            class="tags_container"
          >
            <div>
              #travel
            </div>
            <div>
              #aesthetics
            </div>
            <div>
              #life
            </div>
            <div>
              #blog
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal_wrapper"
    />
    <div
      class="modal_container"
    >
      <img
        alt="current_post"
        class="post_photo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="400"
        src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
        style="color: transparent;"
        width="450"
      />
      <div
        class="post_data_container"
      >
        <img
          alt="close_icon"
          class="close_icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="location"
        >
          Vancouver
        </div>
        <div
          class="date"
        >
          14th October 2023
        </div>
        <div
          class="post_descripton"
        >
          Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit, sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi
        </div>
        <div
          class="tags_container"
        >
          <div>
            #travel
          </div>
          <div>
            #aesthetics
          </div>
          <div>
            #life
          </div>
          <div>
            #blog
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
